.github/workflows/integration-test.yaml

```diff
@@ -0,0 +1,63 @@
+name: Integration tests
+
+on:
+  workflow_call:
+
+jobs:
+  integration-test:
+    runs-on: [self-hosted, linux, X64, xlarge, jammy]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      
+      - name: Fetch Charm Under Test
+        uses: actions/download-artifact@v4
+        with:
+          name: built-charm
+          path: built/
+      
+      - name: Get Charm Under Test Path
+        id: charm-path
+        run: echo "charm_path=$(find built/ -name '*.charm' -type f -print)" >> $GITHUB_OUTPUT
+      
+      - name: Fetch KV Requirer Charm
+        uses: actions/download-artifact@v4
+        with:
+          name: kv-requirer-charm
+          path: kv-requirer/
+      
+      - name: Get KV Requirer Charm Path
+        id: kv-requirer-charm-path
+        run: echo "charm_path=$(find kv-requirer/ -name '*.charm' -type f -print)" >> $GITHUB_OUTPUT
+
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: microk8s
+          channel: 1.29-strict/stable
+          juju-channel: 3.4/stable
+          lxd-channel: 5.20/stable
+  
+      - name: Enable Metallb
+        run: /usr/bin/sg snap_microk8s -c "sudo microk8s enable metallb:10.0.0.2-10.0.0.10"
```

Comment:
This might become flaky if there are network changes. I suggest you use the IP range resulting from `IPADDR=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')`

Reply:
Fixed in #499